src/radical/pilot/utils/component.py

```diff
@@ -1263,7 +1268,10 @@ class AgentComponent(BaseComponent):
     def advance(self, things, state=None, publish=True, push=False, qname=None,
                       ts=None, fwd=True, prof=True):
 
+        things = ru.as_list(things)
```

Comment:
this one is a duplication, we convert it into list in L1278

Reply:
Thanks, cleaned up.